### PR TITLE
Add RFC template

### PR DESCRIPTION
**Pull request by ericclemmons**
As I was working on https://github.com/aws-amplify/amplify-js/issues/4235, I would've benefited from a template based on prior RFCs:

- https://github.com/aws-amplify/amplify-js/issues/3365
- https://github.com/aws-amplify/amplify-js/issues/3279
- https://github.com/aws-amplify/amplify-js/issues/2624

(There are others, but they didn't follow a common format)

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ashika01** [comment]
Love this @ericclemmons. 🎉 
Just couple of thoughts:
- I think in the **Proposed Solution** bit,  we should have a high level architecture and the detailed design and its sample implementation or at least enough context / sample for anyone with familiarity to implement this.
- Also we need to add a section to discuss pros and cons for the solution, compare and contrast the existing solution and solution's comparison with industry practices _(not sure how to name this section)_.

**Comment by ericclemmons** [comment]
You're spot-on, @ashika01. I'll incorporate your feedback.

Whether it's an RFC or task, I'm a big fan of laying out the game-plan for _anyone_ to execute.  It's more effort upfront, but resolving that ambiguity early can make the difference in even _needing_ the feature or not 🤞

File: .github/ISSUE_TEMPLATE/rfc.md [hunk 1]
```diff
@@ -8,17 +8,30 @@ assignees: ''
...
-## Proposed Solution
+## Motivation
 
-<!-- Illustrate the changes you would like to see. Code samples & rationale for changes help! -->
+<!-- Why is this RFC needed? What will happen if accepted? And what would happen if it *isn't* accepted? -->
+
+## Detailed Design
+
+<!-- Provide enough detail on *how* this should be implemented such that someone other than yourself could build it. -->
+<!-- Include examples of how the implementation is used -->
```
Comment (ashika01) [inline .github/ISSUE_TEMPLATE/rfc.md]: Could we move the basic example after Motivation for cohesion.

File: .github/ISSUE_TEMPLATE/rfc.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+_This issue is a Request For Comments (RFC). It is intended to elicit community feedback regarding a proposed change to the library. Please feel free to post comments or questions here._
+
+## Summary
+
+<!-- In one or two sentences, why should this change exist? -->
+
+## Basic Example
+
+<!-- If the RFC involves a new or changed API, include a basic code example. (Omit of not applicable) -->
```
Comment (Amplifiyer) [inline .github/ISSUE_TEMPLATE/rfc.md]: Minor suggestion
```suggestion
<!-- If the RFC involves a new or changed API, include a basic code example. (Omit if not applicable) -->
```

File: .github/ISSUE_TEMPLATE/rfc.md (hunk 1 above)
Comment (ericclemmons) [reply .github/ISSUE_TEMPLATE/rfc.md]: Totally.  I had it here to follow React's example for not "burying the lede", but your suggestion fits better with our typical writing structure.

File: .github/ISSUE_TEMPLATE/rfc.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+title: 'RFC: PROPOSAL HEADLINE'
+labels: feature-request
+assignees: ''
+---
+
+_This issue is a Request For Comments (RFC). It is intended to elicit community feedback regarding a proposed change to the library. Please feel free to post comments or questions here._
+
+## Summary
+
+<!-- In one or two sentences, why should this change exist? -->
```
Comment (sammartinez) [inline .github/ISSUE_TEMPLATE/rfc.md]: Can we remove the `<!-- -->` around all comments so they exist in the template? If you click the preview none of the callouts will be present.

Comment (ericclemmons) [reply .github/ISSUE_TEMPLATE/rfc.md]: Great call.  I've seen instructions both in the template & in comments, but explicit is better.

**Comment by sammartinez** [comment]
> Overall, this is great! One callout that is critical for this PR. **I just want to see about adding a note that makes it clear if any of the pieces of the structure is missing, we will close the RFC and ask the customer to open a feature request**

LGTM. Just want to make sure we have the note about, "If we do not have all sections filled out, this RFC will be closed" Or something to that nature.

**Comment by ericclemmons** [comment]
The wording seems a bit harsh. I think the best thing could be to clarify the difference between this & https://github.com/ericclemmons/amplify-js/blob/github-rfc-template/.github/ISSUE_TEMPLATE/feature_request.md, so that users can make a clear choice.

Honestly, the RFC template seems like the feature request template on steroids, but what I appreciate about the feature request is that it has a low bar for "I wish Amplify did X because of Y".  (RFCs generally have a higher bar for larger, impactful changes, and is closer to what we would write for internal review).

What are your thoughts on solving the distinction issue on https://github.com/aws-amplify/amplify-js/issues/new/choose?

**Comment by ashika01** [comment]
@ericclemmons:  @sammartinez and I talked briefly about this. I think we should merge ur template for now and encourage people to submit RFC and mentor to write better RFC. We will have a WIKI page on life cycle of RFC and Acceptance Criteria.

File: .github/ISSUE_TEMPLATE/rfc.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+---
+name: Request For Comments (RFC)
+about: Gather community feedback regarding a proposed change to the library
+title: 'RFC: PROPOSAL HEADLINE'
+labels: feature-request
+assignees: ''
```
Comment (haverchuck) [inline .github/ISSUE_TEMPLATE/rfc.md]: When I tested this template, the lack of a line break between 'assignees' and '---' caused the text to be XL.

Comment (ericclemmons) [reply .github/ISSUE_TEMPLATE/rfc.md]: Ah! Of course!

Comment (haverchuck) [reply .github/ISSUE_TEMPLATE/rfc.md]: Could just be an artifact introduced by me copying and pasting this file I suppose.  Just something to be aware of.

**Comment by sammartinez** [comment]
> The wording could come across a bit harsh. I think the best thing could be to clarify the difference between this & https://github.com/ericclemmons/amplify-js/blob/github-rfc-template/.github/ISSUE_TEMPLATE/feature_request.md, so that users can make the correct choice.
> 
> Honestly, the RFC template seems like the feature request template on steroids, but what I appreciate about the feature request is that it has a low bar for "I wish Amplify did X because of Y". (RFCs generally have a higher bar for larger, impactful changes, and is closer to what we would write for internal review).
> 
> What are your thoughts on solving the distinction issue on https://github.com/aws-amplify/amplify-js/issues/new/choose?

Understood and part of it is customers may end up just removing the RFC template in general for a feature request tag. Im fine to get this in.

**Comment by ericclemmons** [comment]
Thanks for the input everyone!  Let's see how this works out in the wild and adjust from there.

In the meantime, this should help us have thorough RFCs in the open for customers to help provide feedback on. 🙏

**Pull request merged**